### PR TITLE
Fix crash during onCreate()

### PR DESCRIPTION
**Pull request by sansrivastav**
AB#2690156

LTW reported following crashes and this PR addresses the first 2 crashes:

Stack Trace
1.
Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'android.os.Parcelable android.os.Bundle.getParcelable(java.lang.String)' on a null object reference
at com.microsoft.identity.common.internal.broker.InstallCertActivityLauncher.onCreate(InstallCertActivityLauncher.java:16)

2.
Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'android.os.Parcelable android.os.Bundle.getParcelable(java.lang.String)' on a null object reference
at com.microsoft.identity.common.internal.broker.BrokerActivity.onCreate(BrokerActivity.java:16)

3.
Caused by: java.lang.IllegalStateException: Unexpected fragment type at com.microsoft.identity.common.internal.providers.oauth2.CurrentTaskAuthorizationActivity.onCreate(CurrentTaskAuthorizationActivity.java:142)

See this bug for additional details: https://identitydivision.visualstudio.com/Engineering/_workitems/edit/2690156

### CODE REVIEW

**Comment by shahzaibj** [comment]
> AB#2690156

Please add a proper description. PR descriptions should include a summary of What, Why, and How i.e. what the PR wants to accomplish, why the change is needed, and small summary of the changes made to accomplish the task.

File: common/src/main/java/com/microsoft/identity/common/internal/broker/BrokerActivity.java [hunk 1]
```diff
@@ -62,7 +62,10 @@ protected void onCreate(final Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
 
         if (savedInstanceState == null) {
-            mBrokerInteractiveRequestIntent = getIntent().getExtras().getParcelable(BROKER_INTENT);
+            Bundle extras = getIntent().getExtras();
+            if(extras != null) {
+                mBrokerInteractiveRequestIntent = extras.getParcelable(BROKER_INTENT);
```
Comment (iamgusain) [inline common/src/main/java/com/microsoft/identity/common/internal/broker/BrokerActivity.java:68]: Do we understand in what scenarios the BROKER_INTENT is not being passed?
Should we return an error or ignore silently?

File: common/src/main/java/com/microsoft/identity/common/internal/broker/InstallCertActivityLauncher.java [hunk 2]
```diff
@@ -83,7 +83,10 @@ public final class InstallCertActivityLauncher extends AppCompatActivity {
     protected void onCreate(final Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         if (savedInstanceState == null) {
-            mInstallCertificateIntent = getIntent().getExtras().getParcelable(INSTALL_CERT_INTENT);
+            Bundle extras = getIntent().getExtras();
+            if(extras != null) {
+                mInstallCertificateIntent = extras.getParcelable(INSTALL_CERT_INTENT);
```
Comment (iamgusain) [inline common/src/main/java/com/microsoft/identity/common/internal/broker/InstallCertActivityLauncher.java:89]: Same question here, in what scenarios InstallCertActivity is being launched without the INSTALL_CERT_INTENT, and should we throw an error back?

File: common/src/main/java/com/microsoft/identity/common/internal/broker/BrokerActivity.java (hunk 1 above)
Comment (sansrivastav) [reply common/src/main/java/com/microsoft/identity/common/internal/broker/BrokerActivity.java:68]: There is not much information available except for the crash stack. The partner also has closed the investigation due to low hits.
Even though it is good to understand the reason, we should prevent crashes by adding this check for sure.
I do not think we can return error here, but logging may be helpful.

File: common/src/main/java/com/microsoft/identity/common/internal/broker/InstallCertActivityLauncher.java (hunk 2 above)
Comment (sansrivastav) [reply common/src/main/java/com/microsoft/identity/common/internal/broker/InstallCertActivityLauncher.java:89]: There is not much information available except for the crash stack. The partner also has closed the investigation due to low hits.
Even though it is good to understand the reason, we should prevent crashes by adding this check for sure.
I do not think we can return error here, but logging may be helpful.

File: common/src/main/java/com/microsoft/identity/common/internal/broker/BrokerActivity.java (hunk 1 above)
Comment (shahzaibj) [reply common/src/main/java/com/microsoft/identity/common/internal/broker/BrokerActivity.java:68]: If we don't return error here then what happens? What is the end user experience in this case?

File: common/src/main/java/com/microsoft/identity/common/internal/broker/InstallCertActivityLauncher.java (hunk 2 above)
Comment (shahzaibj) [reply common/src/main/java/com/microsoft/identity/common/internal/broker/InstallCertActivityLauncher.java:89]: If we don't return error here then what happens? What is the end user experience in this case?

**Review by shahzaibj — APPROVED** [review]

**Review by iamgusain — APPROVED** [review]

**Comment by sansrivastav** [comment]
Only Linux broker tests on consumers pipeline is failing 

image

**Comment by tanmaymanolkar1** [comment]
have mentioned @sansrivastav to add the skip consumer check label, since this is a known issue and the only tests which are failing are for Linux.
https://identitydivision.visualstudio.com/Engineering/_build/results?buildId=1191967&view=results
image

**Pull request merged**